### PR TITLE
feat(validators): Add a pending state to AbstractControl

### PR DESCRIPTION
**Pull request by TedSander**
Add a pending state to AbstractControl and a function to set that state on
themselves and their parents. This will be used for both individual async
validators and when the imperitive mode is used. [Design Doc](https://docs.google.com/document/d/1EnJ3-_iFpVKFz1ifN1LkXSGQ7h3A72OQGry2g8eo7IA/edit?pli=1#heading=h.j53rt81eegm4)

### CODE REVIEW

**Comment by TedSander** [comment]
@mhevery @vsavkin This changes the public API to add a pending state to AbstractControls as Victor and I talked about. Please take a detailed look and ensure that the API changes I have are acceptable. Feedback appreciated.

**Comment by TedSander** [comment]
@vsavkin Any comment on this PR?

File: modules/angular2/src/core/forms/model.ts [hunk 1]
```diff
@@ -87,6 +94,15 @@ export class AbstractControl {
     }
   }
 
+  markAsPending({onlySelf}: {onlySelf?: boolean} = {}): void {
+    onlySelf = normalizeBool(onlySelf);
+    this._status = PENDING;
+
+    if (isPresent(this._parent) && !onlySelf) {
+      this._parent.markAsPending({onlySelf: onlySelf});
```
Comment (gkalpak) [inline modules/angular2/src/core/forms/model.ts]: Isn't this the same as `this._parent.markAsPending()` ?

**Comment by vsavkin** [comment]
Merged c9fba3fa1f96a762d9ac43dfba0b40c9f9c4ed71

**Pull request closed without merging**